### PR TITLE
(#171) IdentityToken

### PR DESCRIPTION
**Pull request by llorllale**
This PR is for #171:
* Implemented `IdentityToken`
* Left puzzle to finish tying it all together - the `Auth`s, the `AuthHttpClient`, and the user
* git-ignored a Netbeans-specific folder just so I don't have to keep manually sidestepping it :)

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#188](https://github.com/amihaiemil/docker-java-api/issues/188) is now in scope, role is `REV`

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) everybody who has role `REV` is banned at [#188](https://github.com/amihaiemil/docker-java-api/issues/188); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

File: src/main/java/com/amihaiemil/docker/IdentityToken.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+package com.amihaiemil.docker;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.function.Supplier;
+import javax.json.Json;
+
+/**
+ * A {@link Auth} holding an Identity token.
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/docker/IdentityToken.java]: @llorllale it should be "An Auth"

File: src/test/java/com/amihaiemil/docker/IdentityTokenTestCase.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import javax.json.Json;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Unit tests for {@link IdentityToken}.
+ *
+ * @author George Aristy (george.aristy@gmail.com)
+ * @since 0.0.4
```
Comment (amihaiemil) [inline src/test/java/com/amihaiemil/docker/IdentityTokenTestCase.java]: @llorllale here we should add the ``@version $Id$`` tag

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) everybody who has role `REV` is banned at [#188](https://github.com/amihaiemil/docker-java-api/issues/188); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) everybody who has role `REV` is banned at [#188](https://github.com/amihaiemil/docker-java-api/issues/188); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by 0crat** [comment]
Job `gh:amihaiemil/docker-java-api#188` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#188](https://github.com/amihaiemil/docker-java-api/issues/188) is now out of scope

**Pull request merged**